src/languageserver/handlers/settingsHandlers.ts

```diff
@@ -116,18 +116,17 @@ export class SettingsHandler {
    * AND the schema store setting is enabled. If the schema store setting
    * is not enabled we need to clear the schemas.
    */
-  public setSchemaStoreSettingsIfNotSet(): void {
+  public async setSchemaStoreSettingsIfNotSet(): Promise<void> {
     const schemaStoreIsSet = this.yamlSettings.schemaStoreSettings.length !== 0;
 
     if (this.yamlSettings.schemaStoreEnabled && !schemaStoreIsSet) {
-      this.getSchemaStoreMatchingSchemas()
-        .then((schemaStore) => {
-          this.yamlSettings.schemaStoreSettings = schemaStore.schemas;
-          this.updateConfiguration();
-        })
-        .catch(() => {
-          // ignore
-        });
+      try {
+        const schemaStore = await this.getSchemaStoreMatchingSchemas();
```

Comment:
I do not see where you are stubbing this promise in the test.

Reply:
Because I don't stub it, I stub `xhr` function https://github.com/redhat-developer/yaml-language-server/pull/417/files#diff-9c120b6ffef26acf5a9470c86608e41b18520b4ebdb317aded945ab236739631R40 which used in `getSchemaStoreMatchingSchemas` as the reasl code which I want to test is inside `getSchemaStoreMatchingSchemas`.

Reply:
I was wondering if that might be the case. Thanks for clearing that up.